drew/event_emitter.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+const fs = require('fs');
+const ee = new (require('events').EventEmitter)
+
+var reader = module.exports = function(cb) {
+  let files = [];
+  ee.on('file-one-read', (data) => {
+    var oneData = data.toString();
+    var bufferOneData = new Buffer(oneData);
+    var bufferOneDataStringed = bufferOneData.toString('hex');
+    var bufferOneDataStringedSliced = bufferOneDataStringed.slice(0,8);
+    console.log('the first 8 bytes of file one read: ', bufferOneDataStringedSliced);
+    files.push(bufferOneDataStringedSliced)
+    fs.readFile('./two.txt', (err,data) => {
+      ee.emit('file-two-read', data)
+    })
+  })
+
+  fs.readFile('./one.txt', (err,data) => {
+    ee.emit('file-one-read', data)
+  })
+
```

Comment:
Good job refactoring this Drew.  This should work.  Another way to do this would be to put all of the data manipulation and logging for each file inside of the function for each file read.  Then emit an event to start the next file read data manipulation and so on.